From foo@baz Tue Apr  9 12:12:43 2002
Date: Fri, 2 Nov 2007 16:19:59 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: firmware: remove firmware_(un)register()

These functions are no longer called or needed, so we can remove them.

As I rewrote the whole firmware.c file, add my copyright.

Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/firmware.c |   19 ++-----------------
 include/linux/device.h  |    5 -----
 2 files changed, 2 insertions(+), 22 deletions(-)

--- a/drivers/base/firmware.c
+++ b/drivers/base/firmware.c
@@ -3,11 +3,11 @@
  *
  * Copyright (c) 2002-3 Patrick Mochel
  * Copyright (c) 2002-3 Open Source Development Labs
+ * Copyright (c) 2007 Greg Kroah-Hartman <gregkh@suse.de>
+ * Copyright (c) 2007 Novell Inc.
  *
  * This file is released under the GPLv2
- *
  */
-
 #include <linux/kobject.h>
 #include <linux/module.h>
 #include <linux/init.h>
@@ -18,18 +18,6 @@
 struct kset *firmware_kset;
 EXPORT_SYMBOL_GPL(firmware_kset);
 
-int firmware_register(struct kset *s)
-{
-	s->kobj.kset = firmware_kset;
-	s->kobj.ktype = NULL;
-	return subsystem_register(s);
-}
-
-void firmware_unregister(struct kset *s)
-{
-	subsystem_unregister(s);
-}
-
 int __init firmware_init(void)
 {
 	firmware_kset = kset_create_and_register("firmware", NULL, NULL);
@@ -37,6 +25,3 @@ int __init firmware_init(void)
 		return -ENOMEM;
 	return 0;
 }
-
-EXPORT_SYMBOL_GPL(firmware_register);
-EXPORT_SYMBOL_GPL(firmware_unregister);
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -541,11 +541,6 @@ extern void device_shutdown(void);
 /* drivers/base/sys.c */
 extern void sysdev_shutdown(void);
 
-
-/* drivers/base/firmware.c */
-extern int __must_check firmware_register(struct kset *);
-extern void firmware_unregister(struct kset *);
-
 /* debugging and troubleshooting/diagnostic helpers. */
 extern const char *dev_driver_string(struct device *dev);
 #define dev_printk(level, dev, format, arg...)	\
